src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +35,12 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(i -> i % 2 != 0 ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(e -> e % 2 != 0)
+                .average()
+                .getAsDouble();
```

Comment:
```suggestion
               .orElseThrow(() -> new NoSuchElementException(
                        "Can't get average value from list" + numbers));
```
IDK if in this case value could be null, but it's better to use this method with Optianl